Automattic/jetpack
# Filter post content in Open Graph meta tags

Fixes https://github.com/Automattic/wp-calypso/issues/46928

#### Changes proposed in this Pull Request:

* A short excerpt of post content is added to the `og:description` Open Graph meta tag. Unless an excerpt is supplied by the user, this will be a truncated section of the post body with some processing to remove shortcodes/tags/etc. This PR applies `the_content` filter to the excerpt in order to prevent premium content from being exposed through the meta tag.


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






Test setup:
* Create a new post and insert a Premium Content block at the top of the post.
* View the post on the frontend and open the inspector. Search for the `og:description` tag.

Test cases:
* When viewing the post in an incognito window, verify that the description tag includes only the LoggedOutView text from the Premium Content block
<img width="629" alt="Screen Shot 2021-02-01 at 10 13 42 AM" src="https://user-images.githubusercontent.com/63313398/106500433-ad385200-6476-11eb-8bc2-7c68c08ae561.png">

* When viewing as an admin, the subscriber content is visible in the tag
<img width="576" alt="Screen Shot 2021-02-01 at 10 14 07 AM" src="https://user-images.githubusercontent.com/63313398/106500407-a27dbd00-6476-11eb-96dd-d595214ba04f.png">

* When viewing as a subscriber, the subscriber content is visible in the tag (same as above)




#### Proposed changelog entry for your changes:


*
